crypto/mem.c

```diff
@@ -191,7 +192,9 @@ void *OPENSSL_realloc(void *orig_ptr, size_t new_size) {
   if (orig_ptr == NULL) {
     return OPENSSL_malloc(new_size);
   }
-
+  if (OPENSSL_memory_realloc != NULL) {
+    return OPENSSL_memory_realloc(orig_ptr, new_size);
```

Comment:
should we
```suggestion
    assert(OPENSSL_memory_free != NULL);
    assert(OPENSSL_memory_get_size != NULL);
    return OPENSSL_memory_realloc(orig_ptr, new_size);
```
as in `OPENSSL_malloc`?

Reply:
I don't think it's necessary. I believe the reason OPENSSL_malloc checks for the other 2 is they are required. If a customer is taking control of alloc we can't use our free as it expects the size in the prefix. So If you want to control the memory you need free and size. OPENSSL_realloc can play nicely with our alloc or a custom one. The only people that might be interested in providing a realloc would do it to try to save on allocations (and take on a lot of risk and potential bugs) but might not care about alloc.

Reply:
That is a good point. So defining a custom realloc requires a custom malloc, but not vice versa.